catalyst-uploader.go

```diff
@@ -27,27 +29,60 @@ func run() int {
 		glog.Error(err)
 		return 1
 	}
+	vFlag := flag.Lookup("v")
+	fs := flag.NewFlagSet("catalyst-uploader", flag.ExitOnError)
+
 	// cmd line args
-	describe := flag.Bool("j", false, "Describe supported storage services in JSON format and exit")
-	timeout := flag.Duration("t", 30*time.Second, "Upload timeout")
-	flag.Parse()
+	version := fs.Bool("version", false, "print application version")
+	describe := fs.Bool("j", false, "Describe supported storage services in JSON format and exit")
+	verbosity := fs.String("v", "", "Log verbosity.  {4|5|6}")
+	timeout := fs.Duration("t", 30*time.Second, "Upload timeout")
+	storageBackupURLs := jsonFlag[core.StorageBackupURLs](fs, "storage-backup-urls", `JSON array of {"primary":X,"backup":Y} objects with base storage URLs. If a file fails uploading to one of the primary storages (detected by prefix), it will fallback to the corresponding backup URL after having the prefix replaced`)
```

Comment:
Yeah I thought about creating a custom format here as well, but IMO there are a few disadvantages of using a simpler comma-separated list:
- Writing custom parsing logic instead of just relying on JSON stability
  - e.g. Commas are an allowed value in URLs, so we'd need to be able to escape them somehow as well
- Not being able to configure multiple backup storages anymore without making this new format a little more complex (do we use multiple separators like `,` and `;`?). See [this other comment](https://github.com/livepeer/catalyst-uploader/pull/60#discussion_r1647381707) for more context on supporting multiple storages here.

Adding more than 1 backup would also be easy in the JSON, we'd just need to add a new fields (e.g. `backups` instead of `backup`) then update the config accordingly (or even have a chain of backup URLs: `[{primary:X,backup:Y},{primary:Y,backup:Z},{primary:Z,backup:W},...]` and make the fallback recursive). In both cases (JSON or custom format) we would need code changes to add the support anyway.

As for sending the JSON in the config on the infra I agree it's not the simplest, but I think it wouldn't be too bad. Would be something like:
```yaml
env:
  CATALYST_UPLOADER_STORAGE_BACKUP_URLS: '[{"primary": {{ secrets.CATALYST_RECORDINGS_BUCKET | to_json }}, "backup": {{ secrets.CATALYST_RECORDINGS_BACKUP_BUCKET | to_json }} }]'
```
WDYT?

Reply:
Oh, another alternative here if you think we won't ever need this multiple storage support and the JSON config seems too clumsy would be simply having 2 flags here, one for the single primary URL and one for the single backup. So this would become:
```yaml
env:
  CATALYST_UPLOADER_STORAGE_PRIMARY_URL: {{ secrets.CATALYST_RECORDINGS_BUCKET | quote }}
  CATALYST_UPLOADER_STORAGE_BACKUP_URL: {{ secrets.CATALYST_RECORDINGS_BACKUP_BUCKET | quote }}
```

I still have a _soft_ preference for the current implementation which seems more flexible to me and the code is a little cleaner, but LMK if you prefer to optimize for the simplest interface here.

Reply:
OK I found out that we already have a comma-map of URLs in use on `catalyst-api`, specifically for the [`source-playback-hosts`](https://github.com/livepeer/livepeer-infra/blob/71e1d14745c9db4a1d8e2c2d6229e04703c61eff/values/prod/catalyst.values.yaml#L21) config. The usage is actually [exactly the same](https://github.com/livepeer/catalyst-api/blob/cfed42d7f7bc9406ac62f96e15e7e57383da3449/pipeline/ffmpeg.go#L219-L225)!

Even tho I think there are some theoretical disadvantages with escaping those URLs – which we are just not doing today – I think the consistency on the format is worth it here. So I'll just migrate to a comma-map here too and rename this config to `storage-primary-to-backup-urls`. Let me know if you'd prefer sth else.

Reply:
Yeah, sounds good. 

If we use comma-map somewhere else, then just use the same for the sake of consistency.